### ISSUE TITLE
0.8.0 release does not build

### ISSUE THREAD

**Issue by brad0** [AOMediaCodec/libavif#233]
Just downloaded the 0.8.0 release and trying to build but it fails like so...

```
/home/ports/pobj/libavif-0.8.0/bin/cc  -I/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/shared -I/usr/local/include -I/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/include -O2 -pipe -DNDEBUG   -Weverything -Werror -Wno-bad-function-cast -Wno-cast-align -Wno-conversion -Wno-covered-switch-default -Wno-disabled-macro-expansion -Wno-documentation -Wno-documentation-unknown-command -Wno-double-promotion -Wno-float-equal -Wno-missing-noreturn -Wno-padded -Wno-sign-conversion -pthread -MD -MT CMakeFiles/avifdec.dir/apps/avifdec.c.o -MF CMakeFiles/avifdec.dir/apps/avifdec.c.o.d -o CMakeFiles/avifdec.dir/apps/avifdec.c.o   -c /home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/avifdec.c
In file included from /home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/avifdec.c:6:
/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/shared/avifjpeg.h:10:88: error: unknown type name 'avifChromaUpsampling'
avifBool avifJPEGWrite(avifImage * avif, const char * outputFilename, int jpegQuality, avifChromaUpsampling chromaUpsampling);
                                                                                       ^
In file included from /home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/avifdec.c:7:
/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/shared/avifpng.h:11:95: error: unknown type name 'avifChromaUpsampling'
avifBool avifPNGWrite(avifImage * avif, const char * outputFilename, uint32_t requestedDepth, avifChromaUpsampling chromaUpsampling);
                                                                                              ^
/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/avifdec.c:115:5: error: use of undeclared identifier 'avifChromaUpsampling'
    avifChromaUpsampling chromaUpsampling = AVIF_CHROMA_UPSAMPLING_BILINEAR;
    ^
/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/avifdec.c:160:17: error: use of undeclared identifier 'chromaUpsampling'
                chromaUpsampling = AVIF_CHROMA_UPSAMPLING_BILINEAR;
                ^
/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/avifdec.c:160:36: error: use of undeclared identifier 'AVIF_CHROMA_UPSAMPLING_BILINEAR'
                chromaUpsampling = AVIF_CHROMA_UPSAMPLING_BILINEAR;
                                   ^
/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/avifdec.c:162:17: error: use of undeclared identifier 'chromaUpsampling'
                chromaUpsampling = AVIF_CHROMA_UPSAMPLING_NEAREST;
                ^
/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/avifdec.c:162:36: error: use of undeclared identifier 'AVIF_CHROMA_UPSAMPLING_NEAREST'
                chromaUpsampling = AVIF_CHROMA_UPSAMPLING_NEAREST;
                                   ^
/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/avifdec.c:250:67: error: use of undeclared identifier 'chromaUpsampling'
            if (!avifJPEGWrite(avif, outputFilename, jpegQuality, chromaUpsampling)) {
                                                                  ^
/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/avifdec.c:254:69: error: use of undeclared identifier 'chromaUpsampling'
            if (!avifPNGWrite(avif, outputFilename, requestedDepth, chromaUpsampling)) {
                                                                    ^
9 errors generated.
ninja: build stopped: subcommand failed.
```

**Comment by joedrago** [comment]
I feel like this is finding an old `avif/avif.h`, maybe? What platform/env are you running? I can try to reproduce.

**Comment by brad0** [comment]
I guess that is a possibility from the looks of the command line..

/home/ports/pobj/libavif-0.8.0/bin/cc  -I/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/apps/shared -I/usr/local/include -I/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/include

I would think that -I/home/ports/pobj/libavif-0.8.0/libavif-0.8.0/include should be before -I/usr/local/include to ensure picking up the newer avif.h header over the older header from the previous release.

Taking the existing command line and swapping the header paths around and sure enough it does build Ok.

OpenBSD -current. I'm the port / package maintainer.

**Comment by brad0** [comment]
So can we have CMake fixed so it puts the header paths in the correct order ensuring the newer header is always licked up first?

**Comment by joedrago** [comment]
I'm not sure how your local setup is done or if I have a mistake in CMake, but can you offer a PR or suggestion in my CMakeLists showing where I am doing this incorrectly?

@cryptomilk I know you have a lot of expertise surrounding these things; any suggestions?

**Comment by cryptomilk** [comment]
I'm unable to reproduce this:

```
/usr/bin/cc  -I/home/asn/workspace/projects/libavif/apps/shared -I/usr/include/libpng16 -I/home/asn/workspace/projects/libavif/include  -O2 -g -DNDEBUG   -Werror -Wall -Wextra -pthread -std=
gnu99 -o CMakeFiles/avifdec.dir/apps/avifdec.c.o   -c /home/asn/workspace/projects/libavif/apps/avifdec.c
```

There is no information which operating system nor which cmake version is used. Also we don't know how cmake was invoked.

**Comment by brad0** [comment]
If you're testing on Linux then I wouldn't be surprised you're not able to reproduce the issue.

I did specify the OS and version. It is at the bottom of my second post. But the version isn't an issue. I can use 6.7 and have the same result.

At the moment we're at CMake 3.17.2, but I've seen this using 3.17.1 and 3.16.2. The version does not appear to matter either.

I can reproduce by simply extracting libavif 0.8.0 and running..

cmake -DAVIF_BUILD_APPS=ON -DAVIF_CODEC_AOM=ON -DAVIF_CODEC_DAV1D=ON
make

Doesn't build.

I can give you a shell on a VM if you want to play around.

**Comment by cryptomilk** [comment]
The question is, if this only happens on OpenBSD and if CMake adds it on OpenBSD.

The only way I see that we pick up `/usr/local/include` is that either something goes wrong with:

```
286 target_include_directories(avif
287                            PUBLIC $<BUILD_INTERFACE:${libavif_SOURCE_DIR}/include>
288                                   $<INSTALL_INTERFACE:include>
```

or CMake adds the path by default on OpenBSD.

`cmake --trace-expand`
and
`cmake --debug-output`
should help you to track it down.

**Comment by brad0** [comment]
Huh?

target_include_directories(avifenc PRIVATE apps/shared ${ZLIB_INCLUDE_DIR} ${PNG_PNG_INCLUDE_DIR} ${JPEG_INCLUDE_DIR})
target_include_directories(avifdec PRIVATE apps/shared ${ZLIB_INCLUDE_DIR} ${PNG_PNG_INCLUDE_DIR} ${JPEG_INCLUDE_DIR})

Right there.

**Comment by cryptomilk** [comment]
Try PR #245

**Comment by brad0** [comment]
Thanks! That indeed does the trick.

**Comment by joedrago** [comment]
I just released `v0.8.1`, which should have these changes.
